lib/rules/named-grid-areas-no-invalid/index.js

```diff
@@ -0,0 +1,131 @@
+// @ts-nocheck
+
+'use strict';
+
+const _ = require('lodash');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+
+const ruleName = 'named-grid-areas-no-invalid';
+
+const messages = ruleMessages(ruleName, {
+	noRows: () => 'no rows defined',
+	notRectangular: () => 'template is not rectangular',
+	notContiguousOrRectangular: (areaNames) =>
+		`The following areas are not contiguous or rectangular ${areaNames.join(', ')}`,
+});
+
+function areasArray(gridTemplateString) {
+	const rows = Array.from(gridTemplateString.matchAll(/["']([^"']*)["']/g)).map(
+		(match) => match[1],
+	);
+
+	return rows.map((row) => _.compact(row.trim().split(' ')));
+}
+
+function columnsPerRow(areas) {
+	return areas.map((row) => row.length);
+}
+
+function isRectangular(areas) {
+	return columnsPerRow(areas).every((val, i, arr) => val === arr[0]);
+}
+
+function isContiguousAndRectangular(areas, name) {
+	const indicesByRow = areas.map((row) => {
+		const indices = [];
+		let idx = row.indexOf(name);
+
+		while (idx !== -1) {
+			indices.push(idx);
+			idx = row.indexOf(name, idx + 1);
+		}
+
+		return indices;
+	});
+
+	for (let i = 0; i < indicesByRow.length; i++) {
+		for (let j = i + 1; j < indicesByRow.length; j++) {
+			if (indicesByRow[i].length === 0 || indicesByRow[j].length === 0) {
+				continue;
+			}
+
+			if (!_.isEqual(indicesByRow[i], indicesByRow[j])) {
+				return false;
+			}
+		}
+	}
+
+	return true;
+}
+
+function namedAreas(areas) {
+	const names = new Set(areas.flat());
+
+	names.delete('.');
+
+	return Array.from(names);
+}
+
+function findNotContiguousOrRectangular(areas) {
+	return Array.from(
+		new Set(namedAreas(areas).filter((name) => !isContiguousAndRectangular(areas, name))),
+	);
+}
+
+function removeComment(value) {
```

Comment:
I didn't know if there was already a way to remove comments from values. If there is, I will use that instead